notebooks/pytorch_sparse_quantized_transfer_learning.ipynb

```diff
@@ -0,0 +1,460 @@
+    "        if train:\n",
+    "            loss.backward()\n",
+    "            optimizer.step()\n",
+    "\n",
+    "        running_loss += loss.item()\n",
+    "\n",
+    "        predictions = outputs.argmax(dim=1)\n",
+    "        total_correct += torch.sum(predictions == labels).item()\n",
+    "        total_predictions += inputs.size(0)\n",
+    "\n",
+    "    loss = running_loss / (step + 1.0)\n",
+    "    accuracy = total_correct / total_predictions\n",
+    "    return loss, accuracy"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 4 - Set Up PyTorch Training Objects\n",
+    "In this step, you will select hyperparameters, a device to train your model with, set up DataLoader objects, a loss function, and optimizer.  All of these variables and objects can be replaced to fit your training flow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from torch.utils.data import DataLoader\n",
+    "from torch.nn import CrossEntropyLoss\n",
+    "from torch.optim import Adam\n",
+    "\n",
+    "# hyperparameters\n",
+    "BATCH_SIZE = 32\n",
+    "\n",
+    "# setup device\n",
+    "device = \"cuda\" if torch.cuda.is_available() else \"cpu\"\n",
+    "model.to(device)\n",
+    "print(f\"Using device: {device}\")\n",
+    "\n",
+    "# setup data loaders\n",
+    "train_loader = DataLoader(\n",
+    "    train_dataset, BATCH_SIZE, shuffle=True, pin_memory=True, num_workers=8\n",
+    ")\n",
+    "val_loader = DataLoader(\n",
+    "    val_dataset, BATCH_SIZE, shuffle=False, pin_memory=True, num_workers=8\n",
+    ")\n",
+    "\n",
+    "# setup loss function and optimizer, LR will be overriden by sparseml\n",
+    "criterion = CrossEntropyLoss()\n",
+    "optimizer = Adam(model.parameters(), lr=8e-3)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 5 - Apply a SparseML Recipe and Prune Model\n",
+    "\n",
+    "To run sparse-quantized transfer learning with SparseML, you will download a transfer learning recipe from SparseZoo and use it to create a `ScheduledModifierManager` object.  This manager will be used to wrap the optimizer object to maintain the pre-optimized model's sparsity structure while learning weights for the new dataset as well as performing quantization aware training.\n",
+    "\n",
+    "You can create SparseML recipes to perform various model pruning schedules, quantization aware training, sparse transfer learning, and more.  If you are using a different model than the default, you will have to modify the recipe  file to match the new target's parameters.\n",
+    "\n",
+    "Finally, using the wrapped optimizer object, you will call the training function to prune your model.\n",
+    "\n",
+    "If the kernel shuts down during training, this may be an out of memory error, to resolve this, try lowering the `batch_size` in the cell above."
```

Comment:
```suggestion
    "If the kernel shuts down during training, this may be an out of memory error; to resolve this, try lowering the `batch_size` in the cell above."
```